firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -226,93 +366,72 @@ public void onClick(DialogInterface dialogInterface, int i) {
     return alertDialog;
   }
 
-  /** Signs in the App Distribution tester. Presents the tester with a Google sign in UI */
-  @NonNull
-  public Task<Void> signInTester() {
-    if (signInTaskCompletionSource != null && !signInTaskCompletionSource.getTask().isComplete()) {
-      signInCancellationSource.cancel();
-    }
-
-    signInCancellationSource = new CancellationTokenSource();
-    signInTaskCompletionSource = new TaskCompletionSource<>(signInCancellationSource.getToken());
-
-    Context context = firebaseApp.getApplicationContext();
-    AlertDialog alertDialog = getSignInAlertDialog(context);
-    alertDialog.show();
-
-    return signInTaskCompletionSource.getTask();
-  }
-
-  private void setSignInTaskCompletionError(FirebaseAppDistributionException e) {
-    if (signInTaskCompletionSource != null && !signInTaskCompletionSource.getTask().isComplete()) {
-      signInTaskCompletionSource.setException(e);
-    }
+  private OnSuccessListener<String> getFidGenerationOnSuccessListener(Context context) {
+    return new OnSuccessListener<String>() {
+      @Override
+      public void onSuccess(String fid) {
+        Uri uri =
+            Uri.parse(
+                String.format(
+                    "https://appdistribution.firebase.google.com/pub/testerapps/%s/installations/%s/buildalerts?appName=%s&packageName=%s",
+                    firebaseApp.getOptions().getApplicationId(),
+                    fid,
+                    getApplicationName(context),
+                    context.getPackageName()));
+        openSignInFlowInBrowser(uri);
+      }
+    };
   }
 
-  /** Returns true if the App Distribution tester is signed in */
-  @NonNull
-  public boolean isTesterSignedIn() {
-    return false;
-  }
+  AppDistributionRelease getLatestReleaseFromClient(
+      String fid, String appId, String apiKey, String authToken) {
+    AppDistributionRelease latestRelease = null;
+    try {
+      AppDistributionRelease retrievedLatestRelease =
+          firebaseAppDistributionTesterApiClient.fetchLatestRelease(fid, appId, apiKey, authToken);
 
-  /** Signs out the App Distribution tester */
-  public void signOutTester() {}
+      Context context = firebaseApp.getApplicationContext();
+      long currentInstalledVersionCode = getInstalledAppVersionCode(context);
 
-  @Override
-  public void onActivityCreated(
-      @NonNull @NotNull Activity activity, @androidx.annotation.Nullable @Nullable Bundle bundle) {
-    Log.d(TAG, "Created activity: " + activity.getClass().getName());
-    // if signinactivity is created, sign-in was succesful
-    if (currentlySigningIn && activity instanceof SignInResultActivity) {
-      currentlySigningIn = false;
-      signInTaskCompletionSource.setResult(null);
+      if (Long.parseLong(retrievedLatestRelease.getBuildVersion()) > currentInstalledVersionCode) {
```

Comment:
I think Jeremy pointed this out during demos, but we probably need to compare the display version **and** the build version.

We can discuss offline how best to handle this